internal/state/dataplane/configuration.go

```diff
@@ -235,6 +240,24 @@ func buildServers(listeners map[string]*graph.Listener) (http, ssl []VirtualServ
 	return httpRules.buildServers(), sslRules.buildServers()
 }
 
+// portPathRules keeps track of hostPathRules per port
+type portPathRules map[v1beta1.PortNumber]*hostPathRules
+
+func (p portPathRules) buildServers() []VirtualServer {
+	serverCount := 0
+	for _, rules := range p {
+		serverCount += len(rules.rulesPerHost) + len(rules.httpsListeners)
```

Comment:
why do we need to `+ len(rules.httpsListeners)`?

would it make sense to have a method ServerCount() in hostPathRules, so that portPathRules doesn't depend on implementation details (fields) of hostPathRules?

Reply:
We need `len(rules.httpsListeners` to account for the wildcard https server blocks. This is the max capacity not the exact capacity.

Sure, I can add the ServerCount() method.

Reply:
I also need to add 1 for the default server block we generate for each port.